department-of-veterans-affairs/vets-api
# 39040 logingov user attrs

## Description of change
This PR adds a new route to the inherited proofing solution: `/inherited_proofing/user_attributes`, this route requires an encoded JWT access token (encoded specifically by login.gov's private key), and requires a user to have started the inherited proofing authentication through `/inherited_proofing/auth`, and thus be given a code that can be returned in the JWT access token.  This code is used to retrieve verified user attributes, encrypt them, and return with encrypted user attributes

## Original issue(s)
department-of-veterans-affairs/va.gov-team#39040

## Things to know about this PR
Steps to test:
- Log in normally
- Navigate to `localhost:3000/inherited_proofing/auth`, take note of the auth code in the parameters to the redirected request to login.gov
- Encode an access token at least the fields: `{ inherited_proofing_auth: <auth_code>, exp: <expiration-time> }`
- Since we don't have the ability to encode an access token with login.gov's private key, navigate to: `lib/inherited_proofing/jwt_decoder.rb`, and set `with_validation = false`
- Since we don't have the ability to decrypt user attributes with login.gov's private key, in `lib/inherited_proofing/user_attributes_encryptor.rb`, set `@public_key` to a temporary public key that you know the private key to
- Send a GET request to `/inherited_proofing/user_attributes` with `Bearer: Authorization <access_token_jwt>` in the header
- Response should be a JSON object with an encrypted string in the data
- Using the temporary private key you created above, in a rails console, perform `JWE.decrypt(<encrypted_user_attributes>, <private_key>)`
